### ISSUE TITLE
Some modifier keys are not released

### ISSUE THREAD

**Issue by Flakebi** [any1/wayvnc#71]
Thanks a lot for this project, it’s so useful!
So, I’m using a weird keyboard layout with multiple layers and all kind of functionality and keys: https://neo-layout.org
It’s the *de* layout in the *neo* variant. I don’t think there is a layout that has these keys, so I need to edit https://github.com/any1/wayvnc/blob/master/src/keyboard.c#L161 and set `.variant = "neo"` to get all keys working (like you described in #61).

When I press the modifier key to get to layer 4 (called Mod 4 on the neo website, it reads “Alt Gr” on my keyboard), I can use the arrow keys as expected. However, after I release the modifier key, the layer is still active and I can’t type characters. Sometimes I can reset this by pressing both Mod 4 keys at the same time twice (which locks and unlocks the layer).

wayvnc log:
```python
# Press Mod 4
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT pressed
# Press and release A (which is Down in layer 4; on us layout, this is the D key)
DEBUG: symbol=Down level=0 code=DOWN pressed
DEBUG: symbol=Down level=0 code=DOWN released
# Release Mod 4 (not registered)
# Press A again (D key on us layout), still registered as Down though it *should be registered as A*
DEBUG: Failed to match level on symbol: a
DEBUG: symbol=a level=0 code=AC03 pressed
DEBUG: Failed to match level on symbol: a
DEBUG: symbol=a level=0 code=AC03 released
DEBUG: Failed to match level on symbol: a
DEBUG: symbol=a level=0 code=AC03 pressed
DEBUG: Failed to match level on symbol: a
DEBUG: symbol=a level=0 code=AC03 released
DEBUG: Failed to match level on symbol: a
DEBUG: symbol=a level=0 code=AC03 pressed
DEBUG: Failed to match level on symbol: a
DEBUG: symbol=a level=0 code=AC03 released
# Press and release both Mod 4 keys to lock and unlock layer 4
DEBUG: symbol=ISO_Level5_Shift level=0 code=MDSW pressed
DEBUG: symbol=ISO_Level5_Lock level=4 code=LSGT released
DEBUG: symbol=ISO_Level5_Shift level=0 code=MDSW released
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT pressed
DEBUG: symbol=ISO_Level5_Lock level=4 code=LSGT released
# Press and release A again (this time correctly not interpreted as Down)
DEBUG: symbol=a level=0 code=AC03 pressed
DEBUG: symbol=a level=0 code=AC03 released
DEBUG: symbol=a level=0 code=AC03 pressed
DEBUG: symbol=a level=0 code=AC03 released
```

`wev` log of pressing and releasing Mod 4 in case it is useful:
```python
[14:     wl_keyboard] key: serial: 339045; time: 133707019; key: 108; state: 1 (pressed)
                      sym: ISO_Level5_Shift (65041), utf8: ''
[14:     wl_keyboard] modifiers: serial: 0; group: 0
                      depressed: 00000020: Mod3
                      latched: 00000000
                      locked: 00000000
[14:     wl_keyboard] key: serial: 339047; time: 133707411; key: 108; state: 0 (released)
                      sym: ISO_Level5_Lock (65043), utf8: ''
[14:     wl_keyboard] modifiers: serial: 0; group: 0
                      depressed: 00000000
                      latched: 00000000
                      locked: 00000000
```

If you want any more information or debug things or if my explanation is incomprehensible, tell me.

**Comment by any1** [comment]
Does it work if you add those symbols to the list of shift keys here: https://github.com/any1/wayvnc/blob/master/src/keyboard.c#L310?

**Comment by Flakebi** [comment]
Yes, that works!
I found one difference between outside and inside vnc. Pressing one of 1, 2, 3, 5, 6, 7 while layer 4 is locked, produces the number inside vnc, while it produces the layer 4 symbol outside vnc. It only happens when the layer is locked (pressing Mod 4 + key produces the correct symbol) and only with these keys, other like 8, 9, 0 or characters are working.

Thinking about it, it seems like the non-working symbols are the ones that cannot be found in layer 1 (s and k which produce ¿ and ¡ also don’t work), while the ones that are in layer 1, like arrow keys, are working.

wayvnc log:
```python
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT pressed
DEBUG: symbol=ISO_Level5_Lock level=3 code=TAB pressed
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT released
DEBUG: symbol=ISO_Level5_Lock level=3 code=TAB released
# Layer 4 is locked
DEBUG: Failed to match level on symbol: ordfeminine
DEBUG: symbol=ordfeminine level=4 code=AE01 pressed
DEBUG: Failed to match level on symbol: ordfeminine
DEBUG: symbol=ordfeminine level=4 code=AE01 released
DEBUG: Failed to match level on symbol: masculine
DEBUG: symbol=masculine level=4 code=AE02 pressed
DEBUG: Failed to match level on symbol: masculine
DEBUG: symbol=masculine level=4 code=AE02 released
DEBUG: Failed to match level on symbol: numerosign
DEBUG: symbol=numerosign level=4 code=AE03 pressed
DEBUG: Failed to match level on symbol: numerosign
DEBUG: symbol=numerosign level=4 code=AE03 released
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT pressed
DEBUG: symbol=ISO_Level5_Lock level=3 code=TAB pressed
DEBUG: symbol=ISO_Level5_Lock level=3 code=TAB released
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT released
# Layer 4 is unlocked
DEBUG: symbol=1 level=0 code=AE01 pressed
DEBUG: symbol=2 level=0 code=AE02 pressed
DEBUG: symbol=1 level=0 code=AE01 released
DEBUG: symbol=3 level=0 code=AE03 pressed
DEBUG: symbol=2 level=0 code=AE02 released
DEBUG: symbol=3 level=0 code=AE03 released
# Mod 4 is pressed
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT pressed
DEBUG: symbol=ordfeminine level=4 code=AE01 pressed
DEBUG: symbol=masculine level=4 code=AE02 pressed
DEBUG: symbol=ordfeminine level=4 code=AE01 released
DEBUG: symbol=numerosign level=4 code=AE03 pressed
DEBUG: symbol=masculine level=4 code=AE02 released
DEBUG: symbol=numerosign level=4 code=AE03 released
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT released
```

For a working symbol:
```python
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT pressed
DEBUG: symbol=ISO_Level5_Lock level=3 code=TAB pressed
DEBUG: symbol=ISO_Level5_Lock level=3 code=TAB released
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT released
# Layer 4 is locked
DEBUG: symbol=KP_Divide level=0 code=KPDV pressed
DEBUG: symbol=KP_Divide level=0 code=KPDV released
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT pressed
DEBUG: symbol=ISO_Level5_Lock level=3 code=TAB pressed
DEBUG: symbol=ISO_Level5_Lock level=3 code=TAB released
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT released
# Layer 4 is unlocked
DEBUG: symbol=9 level=0 code=AE09 pressed
DEBUG: symbol=9 level=0 code=AE09 released
# Mod 4 is pressed
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT pressed
DEBUG: symbol=KP_Divide level=4 code=AE09 pressed
DEBUG: symbol=KP_Divide level=4 code=AE09 released
DEBUG: symbol=ISO_Level5_Shift level=0 code=LSGT released
```

**Comment by any1** [comment]
Fixed by #73
